R/checking.R

```diff
@@ -0,0 +1,56 @@
+#' Check the function arguments of the outbreak simulation functions
+#'
+#' @details If more than one argument is invalid, only a single error is
+#'   thrown with the first invalid argument encountered.
+#'
+#'   Arguments being checked are taken from the parent environment
+#'   ([parent.frame()]) rather than passed via named arguments.
+#'
+#' @return `TRUE` if all the checks pass or an error thrown by a \pkg{checkmate}
+#' `assert_*()` function if one or more of the inputs is invalid.
+#' @keywords internal
+check_outbreak_input <- function() {
+  ## get name of the calling function as a character string
+  func <- deparse(as.list(sys.call(-1))[[1]])
+  func <- match.arg(func, choices = c(
+    "outbreak_setup", "outbreak_step", "outbreak_model", "scenario_sim",
+    "ringbp::outbreak_setup", "ringbp::outbreak_step", "ringbp::outbreak_model",
+    "ringbp::scenario_sim"
+  ))
```

Comment:
Do we really need this? It seems this would work fine without?

That said, did you encounter instances where the `func` would have the `ringbp::` bits? If yes we'll have to make sure to remove them / change the if statements below.

If we are worried that this is brittle we could add some tests to make sure the right errors are triggered when calling from the different functions.

Reply:
that said: i don't think this is worth a ton of effort, since its presumably going to be carved up by API rev?